packages/types/src/extensions/defaultClientConfiguration.ts

```diff
@@ -7,22 +7,31 @@ import { ChecksumConfiguration, getChecksumConfiguration, resolveChecksumRuntime
  */
 export interface DefaultClientConfiguration extends ChecksumConfiguration {}
 
+/**
+ * @deprecated will be removed for implicit type.
+ */
 type GetDefaultConfigurationType = (runtimeConfig: any) => DefaultClientConfiguration;
 
 /**
+ * @deprecated moving to @smithy/smithy-client.
  * @internal
  *
  * Helper function to resolve default client configuration from runtime config
+ *
  */
 export const getDefaultClientConfiguration: GetDefaultConfigurationType = (runtimeConfig: any) => {
```

Comment:
in the move I changed this `any` to a type that can be extended later as you add default extensions.